File: user-service/src/main/java/rs/edu/raf/service/impl/FavouriteRecipientServiceImpl.java
package rs.edu.raf.service.impl;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;
import rs.edu.raf.annotations.GeneratedCrudOperation;
import rs.edu.raf.dto.FavouriteRecipientDTO;
import rs.edu.raf.dto.FavouriteRecipientRequestDTO;
import rs.edu.raf.model.user.Client;
import rs.edu.raf.repository.FavouriteRecipientRepository;
import rs.edu.raf.mapper.FavouriteRecipientMapper;
import rs.edu.raf.security.JwtUtil;
import rs.edu.raf.service.FavouriteRecipientService;

import java.util.List;

@Service
@AllArgsConstructor
public class FavouriteRecipientServiceImpl implements FavouriteRecipientService {

    private FavouriteRecipientRepository favouriteRecipientRepository;
    private FavouriteRecipientMapper favouriteRecipientMapper;
    private JwtUtil jwtUtil;
    @Override
    @GeneratedCrudOperation
    public FavouriteRecipientDTO addFavouriteClient(FavouriteRecipientRequestDTO favouriteRecipientRequestDTO) {
        return favouriteRecipientMapper.favouriteRecipientToFavouriteRecipientDTO(favouriteRecipientRepository.save(favouriteRecipientMapper.favouriteRecipientRequestDTOtoFavouriteRecipient(favouriteRecipientRequestDTO, (Client) jwtUtil.getCurrentUser())));
    }

    @Override
    @GeneratedCrudOperation
    public FavouriteRecipientDTO editFavouriteClient(FavouriteRecipientRequestDTO favouriteRecipientRequestDTO) {
        return favouriteRecipientMapper.favouriteRecipientToFavouriteRecipientDTO(favouriteRecipientRepository.save(favouriteRecipientMapper.favouriteRecipientRequestDTOtoFavouriteRecipient(favouriteRecipientRequestDTO, (Client) jwtUtil.getCurrentUser())));
    }

    @Override
    @GeneratedCrudOperation
    public void deleteFavouriteClient(Long id) {
        favouriteRecipientRepository.deleteById(id);
    }

    @Override
    @GeneratedCrudOperation
    public List<FavouriteRecipientDTO> findFavouriteClientsForClient() {
        return favouriteRecipientRepository.findByClient_Id(jwtUtil.getCurrentUser().getId()).stream().map(favouriteRecipientMapper::favouriteRecipientToFavouriteRecipientDTO).toList();
    }
}

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/service/impl/FavouriteRecipientServiceImpl.kt
package org.example.userservicekotlin.service.impl

import org.example.userservicekotlin.dto.FavouriteRecipientCreateDTO
import org.example.userservicekotlin.dto.FavouriteRecipientDTO
import org.example.userservicekotlin.mapper.FavouriteRecipientMapper
import org.example.userservicekotlin.model.user.Client
import org.example.userservicekotlin.repository.FavouriteRecipientRepository
import org.example.userservicekotlin.security.JwtUtil
import org.example.userservicekotlin.service.FavouriteRecipientService
import org.springframework.stereotype.Service

@Service
class FavouriteRecipientServiceImpl(
    val favouriteRecipientRepository: FavouriteRecipientRepository,
    val favouriteRecipientMapper: FavouriteRecipientMapper,
    val jwtUtil: JwtUtil
) : FavouriteRecipientService {

    override fun addFavouriteClient(favouriteRecipientCreateDTO: FavouriteRecipientCreateDTO): FavouriteRecipientDTO {
        return favouriteRecipientMapper.favouriteRecipientToFavouriteRecipientDTO(
            favouriteRecipientRepository.save(
                favouriteRecipientMapper.favouriteRecipientRequestDTOtoFavouriteRecipient(
                    favouriteRecipientCreateDTO,
                    jwtUtil.getCurrentUser() as Client
                )
            )
        )
    }

    override fun editFavouriteClient(favouriteRecipientCreateDTO: FavouriteRecipientCreateDTO): FavouriteRecipientDTO {
        return favouriteRecipientMapper.favouriteRecipientToFavouriteRecipientDTO(
            favouriteRecipientRepository.save(
                favouriteRecipientMapper.favouriteRecipientRequestDTOtoFavouriteRecipient(
                    favouriteRecipientCreateDTO,
                    jwtUtil.getCurrentUser() as Client
                )
            )
        )
    }

    override fun deleteFavouriteClient(id: Long?) {
        favouriteRecipientRepository.deleteById(id!!)
    }

    override fun findFavouriteClientsForClient(): List<FavouriteRecipientDTO> {
        return favouriteRecipientRepository.findByClient_Id(jwtUtil.getCurrentUser().id!!)
            .map { favouriteRecipient ->
                favouriteRecipientMapper.favouriteRecipientToFavouriteRecipientDTO(
                    favouriteRecipient
                )
            }.toList()
    }
}